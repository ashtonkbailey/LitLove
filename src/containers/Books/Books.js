import React from 'react';

import '../../index.scss';

const Books = () => {
  return (
    <div>
    </div>
  )
}

export default Books;
