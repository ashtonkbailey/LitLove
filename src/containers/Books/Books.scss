.books-text {
  font-family: Sarabun;
  font-size: 30px;
  font-weight: normal;
  text-align: center;
  color: #FFFFFF;
}

hr {
  width: 90%;
  border: 0;
  height: 3px;
  background: #1f1f1f;
  background-image: linear-gradient(to right, #1f1f1f, #dab192, rgba(44, 44, 44, 0.6));
  margin-bottom: 1em;
}

.books-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  padding: 4em;
  padding-top: 1em;
}

.text-div {
  width: 300px;
  padding: 5em;
  padding-top: 11em;
}

.end-text {
  display: block;
  font-family: Sarabun;
  font-size: 30px;
  font-weight: normal;
  text-align: center;
  color: #FFFFFF;
  margin: auto;
  margin-bottom: 1em;
}

.sorry-text {
  font-family: Sarabun;
  font-size: 30px;
  font-weight: normal;
  text-align: center;
  color: #FFFFFF;
  margin: 3em auto;
  height: 500px;
  width: 500px;
}

a {
  color: #dab192;
}