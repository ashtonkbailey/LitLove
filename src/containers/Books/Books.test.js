import React from 'react';
import { shallow } from 'enzyme';
import { Link } from 'react-router-dom';

import { Books, mapStateToProps } from './Books';

describe('Books', () => {
  let wrapper;
  let type;
  let possibleBooks;
  let lovedBook;
  let recommendations;

  it('should match the snapshot if type is confirm', () => {
    type = 'confirm';
    possibleBooks = {items: [{ title: 'hi', author: 'bob loblaw', id: 123 }]};
    lovedBook = { title: 'hi', author: 'bob loblaw', id: 123 };
    recommendations = [{ title: 'hey', author: 'bob loblob', id: 123 }];
    const prompt = (
      <h3 className="end-text">
        Don't see your book? Click
        <span>
          <Link to="/new"> here </Link>
        </span>
        to try again.
      </h3>
    )
    wrapper = shallow(<Books type={type} possibleBooks={possibleBooks} lovedBook={lovedBook} recommendations={recommendations} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot if type is recommendations', () => {
    type = 'recommendations';
    possibleBooks = [{ title: 'hi', author: 'bob loblaw', id: 123 }];
    lovedBook = { title: 'hi', author: 'bob loblaw', id: 123 };
    recommendations = [{ title: 'hey', author: 'bob loblob', id: 123 }];
    wrapper = shallow(<Books type={type} possibleBooks={possibleBooks} lovedBook={lovedBook} recommendations={recommendations} />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    let mockState;
    let expected;
    let result;

    it('should return an object with the possibleBooks array', () => {
      mockState = {
        possibleBooks: [{ title: 'hi', author: 'bob loblaw' }]
      };
      expected = {
        possibleBooks: [{ title: 'hi', author: 'bob loblaw' }]
      };
      result = mapStateToProps(mockState);
      expect(result).toEqual(expected);
    });

    it('should return an object with the lovedBook', () => {
      mockState = {
        lovedBook: { title: 'hi', author: 'bob loblaw' }
      };
      expected = {
        lovedBook: { title: 'hi', author: 'bob loblaw' }
      };
      result = mapStateToProps(mockState);
      expect(result).toEqual(expected);
    });
    
    it('should return an object with the recommendations array', () => {
      mockState = {
        recommendations: [{ title: 'hi', author: 'bob loblaw' }]
      };
      expected = {
        recommendations: [{ title: 'hi', author: 'bob loblaw' }]
      };
      result = mapStateToProps(mockState);
      expect(result).toEqual(expected);
    });  
  });

})
