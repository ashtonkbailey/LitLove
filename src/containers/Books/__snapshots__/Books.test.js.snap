// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Books should match the snapshot if type is confirm 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Books
    lovedBook={
      Object {
        "author": "bob loblaw",
        "title": "hi",
      }
    }
    possibleBooks={
      Object {
        "items": Array [
          Object {
            "author": "bob loblaw",
            "title": "hi",
          },
        ],
      }
    }
    recommendations={
      Array [
        Object {
          "author": "bob loblob",
          "title": "hey",
        },
      ]
    }
    type="confirm"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="books-text"
        >
          Confirm your loved book:
        </h3>,
        <hr />,
        <div
          className="books-container"
        >
          <Connect(BookCard)
            author="bob loblaw"
            title="hi"
            type="confirm"
          />
          <h3
            className="end-text"
          >
            Don't see your book? Click
            <span>
              <Link
                replace={false}
                to="/new"
              >
                 here 
              </Link>
            </span>
            to try again.
          </h3>
        </div>,
      ],
      "className": "books",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Confirm your loved book:",
          "className": "books-text",
        },
        "ref": null,
        "rendered": "Confirm your loved book:",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            Array [
              <Connect(BookCard)
                author="bob loblaw"
                title="hi"
                type="confirm"
              />,
            ],
            <h3
              className="end-text"
            >
              Don't see your book? Click
              <span>
                <Link
                  replace={false}
                  to="/new"
                >
                   here 
                </Link>
              </span>
              to try again.
            </h3>,
          ],
          "className": "books-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "author": "bob loblaw",
              "title": "hi",
              "type": "confirm",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Don't see your book? Click",
                <span>
                  <Link
                    replace={false}
                    to="/new"
                  >
                     here 
                  </Link>
                </span>,
                "to try again.",
              ],
              "className": "end-text",
            },
            "ref": null,
            "rendered": Array [
              "Don't see your book? Click",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    replace={false}
                    to="/new"
                  >
                     here 
                  </Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": " here ",
                    "replace": false,
                    "to": "/new",
                  },
                  "ref": null,
                  "rendered": " here ",
                  "type": [Function],
                },
                "type": "span",
              },
              "to try again.",
            ],
            "type": "h3",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="books-text"
          >
            Confirm your loved book:
          </h3>,
          <hr />,
          <div
            className="books-container"
          >
            <Connect(BookCard)
              author="bob loblaw"
              title="hi"
              type="confirm"
            />
            <h3
              className="end-text"
            >
              Don't see your book? Click
              <span>
                <Link
                  replace={false}
                  to="/new"
                >
                   here 
                </Link>
              </span>
              to try again.
            </h3>
          </div>,
        ],
        "className": "books",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Confirm your loved book:",
            "className": "books-text",
          },
          "ref": null,
          "rendered": "Confirm your loved book:",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              Array [
                <Connect(BookCard)
                  author="bob loblaw"
                  title="hi"
                  type="confirm"
                />,
              ],
              <h3
                className="end-text"
              >
                Don't see your book? Click
                <span>
                  <Link
                    replace={false}
                    to="/new"
                  >
                     here 
                  </Link>
                </span>
                to try again.
              </h3>,
            ],
            "className": "books-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "author": "bob loblaw",
                "title": "hi",
                "type": "confirm",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Don't see your book? Click",
                  <span>
                    <Link
                      replace={false}
                      to="/new"
                    >
                       here 
                    </Link>
                  </span>,
                  "to try again.",
                ],
                "className": "end-text",
              },
              "ref": null,
              "rendered": Array [
                "Don't see your book? Click",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      replace={false}
                      to="/new"
                    >
                       here 
                    </Link>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": " here ",
                      "replace": false,
                      "to": "/new",
                    },
                    "ref": null,
                    "rendered": " here ",
                    "type": [Function],
                  },
                  "type": "span",
                },
                "to try again.",
              ],
              "type": "h3",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Books should match the snapshot if type is recommendations 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Books
    lovedBook={
      Object {
        "author": "bob loblaw",
        "title": "hi",
      }
    }
    possibleBooks={
      Array [
        Object {
          "author": "bob loblaw",
          "title": "hi",
        },
      ]
    }
    recommendations={
      Array [
        Object {
          "author": "bob loblob",
          "title": "hey",
        },
      ]
    }
    type="recommendations"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="books-text"
        >
          Here are some recommendations based on '[object Object]':
        </h3>,
        <hr />,
        <div
          className="books-container"
        >
          <Connect(BookCard)
            author="bob loblob"
            title="hey"
            type="recommendations"
          />
        </div>,
      ],
      "className": "books",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Here are some recommendations based on '[object Object]':",
          "className": "books-text",
        },
        "ref": null,
        "rendered": "Here are some recommendations based on '[object Object]':",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            Array [
              <Connect(BookCard)
                author="bob loblob"
                title="hey"
                type="recommendations"
              />,
            ],
            undefined,
          ],
          "className": "books-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "author": "bob loblob",
              "title": "hey",
              "type": "recommendations",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          undefined,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="books-text"
          >
            Here are some recommendations based on '[object Object]':
          </h3>,
          <hr />,
          <div
            className="books-container"
          >
            <Connect(BookCard)
              author="bob loblob"
              title="hey"
              type="recommendations"
            />
          </div>,
        ],
        "className": "books",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Here are some recommendations based on '[object Object]':",
            "className": "books-text",
          },
          "ref": null,
          "rendered": "Here are some recommendations based on '[object Object]':",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              Array [
                <Connect(BookCard)
                  author="bob loblob"
                  title="hey"
                  type="recommendations"
                />,
              ],
              undefined,
            ],
            "className": "books-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "author": "bob loblob",
                "title": "hey",
                "type": "recommendations",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            undefined,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
